contracts/multisig-prover/src/execute.rs

```diff
@@ -126,31 +128,116 @@ pub fn update_worker_set(deps: DepsMut, env: Env) -> Result<Response, ContractEr
         pub_keys.push(pub_key);
     }
 
-    let new_worker_set = WorkerSet::new(snapshot, pub_keys, env.block.height)?;
+    Ok((snapshot, participants.into_iter().zip(pub_keys).collect()))
+}
+
+pub fn update_worker_set(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    let (snapshot, participants) = get_workers_info(&deps, &env)?;
+
+    let new_worker_set = WorkerSet::new(
+        participants.clone(),
+        snapshot.quorum.into(),
+        env.block.height,
+    )?;
+
+    let cur_worker_set = CURRENT_WORKER_SET.may_load(deps.storage)?;
 
-    let cur_worker_set = CURRENT_WORKER_SET.load(deps.storage)?;
+    let key_id = new_worker_set.id();
```

Comment:
key_id = new_worker_set.id()? This reads so weird that we really need to clear our terminologies. I'd rather seeing the `.hash` method on WorkerSet being used here than trying to understand that this isn't a bug.

Reply:
I think the id is used here because CJ was planning to remove the key_id everywhere in the future and use the worker set id instead when needed. Is that so @cjcobb23 ?

Reply:
yes we can remove the key id entirely, and just use a worker set id

Reply:
Task to remove key_id: https://axelarnetwork.atlassian.net/browse/AXE-1852